src/core/server/http/http_service.ts

```diff
@@ -69,7 +75,12 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
 
     const config = await this.config$.pipe(first()).toPromise();
 
-    return this.httpServer.setup(config);
+    const httpSetup = (this.httpServer.setup(config) || {}) as HttpServiceSetup;
```

Comment:
**nit** `Object.assign` handles `undefined` and `null` out of the box, so everything can be expressed as 
```js
return Object.assign({}, this.httpServer.setup(config), {
  createNewServer: ...
});
```

Reply:
@restrry i brought up the null/undefined behavior w/r/t to object.assign last week as well -- https://github.com/elastic/kibana/pull/36804#discussion_r287143358.

however, the issue here is that `this.httpServer.setup(config)` can return `undefined` -- i'm not trying to mix it with another object, I'm just trying to satisfy typescript's check since I can't cast `undefined` nor `null` to `HttpServiceSetup`, but I can cast a empty object to it.

I could switch this to `Object.assign(this.httpServer.setup(config), {}) as HttpServiceSetup)` if we prefer this syntax to using a boolean operator in this case.

Reply:
that's completely optional, you can ignore my comment